[14.0][MIG] sale_layout_category_hide_detail

rousseldenis opened the pull request:
Added:

- [x] Able to hide section line subtotal

ernestotejeda committed:
sale_layout_category_hide_detail: New module

Hide details for sections in sales orders and invoices for reports and
customer portal

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-reporting-11.0/sale-reporting-11.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-11-0/sale-reporting-11-0-sale_layout_category_hide_detail/es/

ernestotejeda committed:
[MIG] sale_layout_category_hide_detail: Migration to 12.0

pedrobaeza committed:
[MIG] sale_layout_category_hide_detail: Migration scripts

sebalix committed:
[IMP] sale_layout_category_hide_detail: black, isort

sebalix committed:
[MIG] sale_layout_category_hide_detail: Migration to 13.0

ernestotejeda committed:
[FIX] sale_layout_category_hide_detail: Propagate 'Show details' to invoice"

Propagate values in 'Show details' and 'Show subtotal' fields from Sale Order Lines to Invoice Lines

TDu committed:
[FIX] sale_layout_category_hide_detail: Column rendering in section + tests + linting + JS for column not in fieldsInfo list

TDu committed:
[IMP] sale_layount_category_hide_detail hidding amounts

Add a third clickable option on sections to hide all amounts in the
detailed lines of a section.

pedrobaeza committed:
[FIX] sale_layout_category_hide_detail: Missing condition

If a note line is put in first position, we get an error due to this missing condition.

oca-travis committed:
[UPD] Update sale_layout_category_hide_detail.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-reporting-13.0/sale-reporting-13.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-13-0/sale-reporting-13-0-sale_layout_category_hide_detail/

OCA-git-bot committed:
[UPD] README.rst

joao-p-marques committed:
Apply pre-commit

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (11 of 11 strings)

Translation: sale-reporting-13.0/sale-reporting-13.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-13-0/sale-reporting-13-0-sale_layout_category_hide_detail/es/

oca-travis committed:
[UPD] Update sale_layout_category_hide_detail.pot

OCA-git-bot committed:
sale_layout_category_hide_detail 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-reporting-13.0/sale-reporting-13.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-13-0/sale-reporting-13-0-sale_layout_category_hide_detail/

rousseldenis committed:
[IMP] sale_layout_category_hide_detail: black, isort, prettier

rousseldenis committed:
[14.0][MIG] sale_layout_category_hide_detail

rousseldenis commented:
/ocabot migration sale_layout_category_hide_detail

rousseldenis commented:
@ernestotejeda This is an interesting widget.

But what do you think about making it more generic (only with the js part), in order to apply it on more apps that have form sections(purchase, ...) ?

ernestotejeda commented:
> But what do you think about making it more generic ...

I think that's a good idea. what do you think @pedrobaeza ?

pedrobaeza commented:
As the widget has been moved from sale to account during previous version (on the initial version of this module it wasn't), we can have an `account_layout_category_hide_detail` with account.move + widget stuff, and then add the sale one on top of it, as well as the purchase can be created easily.

I agree it can be a good move.

ernestotejeda commented on sale_layout_category_hide_detail/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+    "version": "14.0.1.0.0",
+    "category": "Sales Management",
+    "website": "https://github.com/OCA/sale-reporting",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
```
```suggestion
    "author": "Tecnativa, Odoo Community Association (OCA)",
```

ernestotejeda reviewed (commented):
Funcional review 👍🏼 

- I think readme should be improved to add explanation about 'Show line amount' option 
- You could take advantaje of the migration to adapt javascript code to ES, what do you think?

rousseldenis committed:
[14.0][IMP] sale_layout_category_hide_detail: Remove section name in subtotal

The section name on subtotal level introduces bad UX as it can be long.
It is not necessary as it is visible on top of the section.

angelmoya reviewed (commented):
Functional review:

With "line amount hiden", pdf works, but website show details.

rousseldenis committed:
[14.0][IMP] sale_layout_category_hide_detail: Add possibility to hide section line subtotal

rousseldenis commented:
> Functional review:
> 
> With "line amount hiden", pdf works, but website show details.

Indeed. But after check, it was present in previous versions. I'll try to fix it

rousseldenis committed:
[14.0][FIX] sale_layout_category_hide_detail: Make the hide line amount working

rousseldenis commented:
@angelmoya Could you try now ?

ernestotejeda reviewed (commented):
Code review: 👍🏼 
Functional review: 👍🏼 

I remember when I migrated the module to v12 @pedrobaeza suggested me to have the section name in the subtotal: https://github.com/OCA/sale-reporting/pull/70#pullrequestreview-259434580.
@pedrobaeza Is there a specific reason to do that or we  can keep the total without the section name as Rousseldenis has done now for v14?

pedrobaeza commented:
Yes, it's on purpose, and specified by our customer, as when having very big sections, you don't know which section you are putting the subtotal for. Can you please restore it, @rousseldenis, or add a configuration company parameter for specifying the behavior?

rousseldenis commented:
> Yes, it's on purpose, and specified by our customer, as when having very big sections, you don't know which section you are putting the subtotal for. Can you please restore it, @rousseldenis, or add a configuration company parameter for specifying the behavior?

Ok, try to do that. I understand the purpose. But IMHO the presentation is a little bit dirty. I'm not expert in UI. Keep in mind section name can be long and subtotal string can be 'lost in space'...

pedrobaeza commented:
One thing we can do is to put such text in a separate element for not moving `Subtotal` text.

elvise commented:
> One thing we can do is to put such text in a separate element for not moving `Subtotal` text.

Hi @rousseldenis great PR!!!
any plan for get it done ?

elvise commented:
@rousseldenis can we add an option for page break ?

angelmoya commented:
I try Todo page break in this version, and this is not easy to do, beacuse sections are inside table, and div page break do not work inside a table.

To do it, it is required to redefine table and create a table for section, and it's not possible to do inheriting views.

Maybe you can do other way, or maybe page break can work inside a table...

francesco-ooops commented:
Hi @rousseldenis , any plans for this one?

rousseldenis commented:
> Hi @rousseldenis , any plans for this one?

I'll try to get back here today or tomorrow

angelmoya commented:
Functional revision not OK (tested on runbot), on posted invoice clicking on button "Switch to: detail hiden" does visible the "paid" ribbon

![image](https://user-images.githubusercontent.com/6750341/178291962-18e8b1cd-a3f5-42d0-a5af-0169590c8b3d.png)

rousseldenis commented:
> Functional revision not OK (tested on runbot), on posted invoice clicking on button "Switch to: detail hiden" does visible the "paid" ribbon
> 
> ![image](https://user-images.githubusercontent.com/6750341/178291962-18e8b1cd-a3f5-42d0-a5af-0169590c8b3d.png)

Thanks for the review but this module does not take care about that ribbon. Only lines.

ernestotejeda commented:
ping @rousseldenis :)

jeroenev commented:
> Thanks for the review but this module does not take care about that ribbon. Only lines.

If the module breaks default Odoo functionality, I think it's relevant to fix this before finishing this PR
because it does seem like that has an unintended impact on the payment banner

ao-landoo commented:
+1 to the bug, it is marking the invoice as paid when it shouldn't. I've been looking for the fix for a few hours but I cannot find the problem's solution...

ao-landoo commented on sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js:
```diff
@@ -0,0 +1,82 @@
+        _toggleValue: function (event) {
+            event.preventDefault();
+            event.stopPropagation();
+            this._setValue(!this.value);
```
The invoice gets marked as paid when this line is executed

victoralmau replied:
Reviewing this in detail would have a complex solution; however, most importantly, I believe that you should NOT be able to modify anything (use the buttons) if the invoice is not in draft. This change would therefore solve the problem. 
What do you think about it?

jeroenev replied:
If that actually solves the problem then I think it's an acceptable solution

victoralmau commented:
Superseed by https://github.com/OCA/sale-reporting/pull/170
